### ISSUE TITLE
New Rule: (No empty file)

### ISSUE THREAD

**Issue by Donald1597** [eslint/eslint#19385]
### Rule details

This rule checks if a file is empty.

### Related ECMAScript feature

General file content check (no direct ECMAScript feature)

### What type of rule is this?

Warns about a potential problem

### Example code

```js
Examples of **incorrect** code for this rule:

::: incorrect


// Empty file
// File with only whitespace


:::

Examples of **correct** code for this rule:

:::correct


const x = 1;
// Comment
```

### Why should this rule be in the core instead of a plugin?

This rule should be included in the core because:

1. Universal Use Case: Empty files are a common issue across all JavaScript projects, regardless of the framework or library being used. This rule would benefit all developers, not just those with specific use cases.

2. Code Quality: Empty files often indicate development oversights (e.g., forgotten or misnamed files). This rule helps maintain a clean codebase and prevents unnecessary files from being committed.

3. Integration with Existing Rules: ESLint already has rules like `no-unused-vars` and `no-empty` to ensure code quality. This rule naturally complements them by ensuring files are not empty.

4. Simplicity: The rule is straightforward to implement and maintain, with no external dependencies. It doesn't require complex configuration, making it a good fit for the core.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

**Comment by Tanujkanti4441** [comment]
Hi @Donald1597, thanks for the issue.

`eslint-plugin-unicorn` also has a similar rule called [no-empty-file](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-empty-file.md) that disallows files that only have 
    Whitespace
    Comments
    Directives
    Empty statements
    Empty block statements
    Hashbang

Does this rule proposal also aims to disallows with files with these contents or just files with whitespaces?

**Comment by Donald1597** [comment]
Hi @Tanujkanti4441 
only files containing white space, but I can extend it to other cases if necessary

**Comment by Tanujkanti4441** [comment]
>but when I wanted to use eslint-plugin-unicorn with the rule 'unicorn/no-empty-file': 'error', I got an error saying that this rule does not exist. That's why I wanted to add this one

But i tried it locally and not getting any error here is a reproducible example:
https://stackblitz.com/edit/stackblitz-starters-mvptfxwj?file=package.json,eslint.config.mjs,index.js

And if the plugin has rule then i don't see any other reason to add the rule in ESLint core.

**Comment by Donald1597** [comment]
Ok no problem. Thks

**Comment by mdjermanovic** [comment]
Looks like there's an agreement to close this issue since this rule already exists in a plugin.
